ApplicationLogs/LogReader.cs

```diff
@@ -52,7 +52,7 @@ public void OnPersist(Snapshot snapshot, IReadOnlyList<Blockchain.ApplicationExe
             foreach (var appExec in applicationExecutedList)
             {
                 JObject json = new JObject();
-                json["txid"] = appExec.Transaction.Hash.ToString();
+                json["txid"] = appExec.Transaction?.Hash.ToString();
```

Comment:
```suggestion
                json["type"] = appExec.Transaction == null ? "block" : "tx";
                json["txid"] = appExec.Transaction?.Hash.ToString();
```

Reply:
We can check the `trigger` for knowing the type.

Reply:
The unique without Tx is System, right?

Reply:
Yes.